libraries/AP_Compass/AP_Compass.cpp

```diff
@@ -453,6 +456,12 @@ void Compass::_detect_backends(void)
     }
 #elif HAL_COMPASS_DEFAULT == HAL_COMPASS_QFLIGHT
     _add_backend(AP_Compass_QFLIGHT::detect(*this));
+#elif CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_MINLURE
+    _add_backend(AP_Compass_HMC5843::probe_mpu6000(*this));
+    _add_backend(AP_Compass_HMC5843::probe(*this,
+                                           Linux::I2CDeviceManager::from(hal.i2c_mgr)->get_device({ "platform/80860F41:05", "pci0000:00/0000:00:18.6" },
```

Comment:
I don't like the redundancy on those paths (and future use of I2C device abstraction on other parts). (https://github.com/diydrones/ardupilot/pull/3792/commits/48e18da7aa2676a4cf0872b664367b77792384f9#diff-6c1cc10ba680a71a3e4600a38e243127R56)

But I think using a define isn't pretty either...